core/ibft.go

```diff
@@ -708,7 +728,7 @@ func (i *IBFT) validateProposal(msg *proto.Message, view *proto.View) bool {
 	)
 
 	for _, tuple := range roundsAndPreparedBlockHashes {
-		if tuple.round > maxRound {
+		if tuple.round >= maxRound {
```

Comment:
You have not included a test for this scenario, for example:
>A test scenario for this would be to validate a proposal for which RC messages have a PC corresponding to round 0 - this is a valid situation because you can be on for ex. round 100, if the last finalized prepare you did (the proposal for which you received Q - 1 messages) was for round 0, this loop won’t catch that, but it should

Reply:
Resolving as @trimixlover mentioned it will be tackled in a separate PR